std/stdio.d

```diff
@@ -1522,6 +1524,142 @@ for every line.
     }
 
 /**
+Creates a file with a randomly generated name and returns an open File to it.
+
+There is nothing special about the file created other than the fact that its
+name is randomly generated, and it defaults to being in the temp directory on
+the system. It will only be deleted if explicitly deleted or if it's in the
+temp directory, and the temp directory is cleared out (as happens on system
+startup or shutdown on some systems).
+
+The file is created with R/W permissions and opened with $(D "w+b"). On POSIX
+systems, the permissions are restricted to the current user, though the
+effective permissions are modified by the process' umask in the usual way.
+
+Params:
+    prefix = Prefix for the generated file name.
+    suffix = Suffix for the generated file name (which also provides a way to
+             give the file an extension).
+    dir    = Directory of the temporary file. Defaults to the result of
+             $(XREF file, tempDir).
+
+Throws:
+    $(D ErrnoException) if the file could not be created.
+  */
+    static File tempFile(string prefix, string suffix, string dir)
+    {
+        import std.exception : ErrnoException;
+        import std.file : isSymlink;
+        import std.path : absolutePath, baseName, buildPath, dirName;
+
+        static string genTempName(string prefix, string suffix, string dir)
+        {
+            import std.ascii : digits, letters;
+            import std.random : rndGen;
+            import std.range : chain;
+            auto name = new char[](prefix.length + 15 + suffix.length);
+            name[0 .. prefix.length] = prefix;
+            name[$ - suffix.length .. $] = suffix;
+
+            rndGen.popFront();
+            auto chars = chain(cast(immutable(ubyte[]))letters, cast(immutable(ubyte[]))digits);
+            foreach(ref c; name[prefix.length .. $ - suffix.length])
+            {
+                c = chars[rndGen.front % chars.length];
+                rndGen.popFront();
+            }
+
+            return buildPath(dir, name);
+        }
+
+        // Once should be enough given the randomness, but we'll give it three
+        // times to be safe.
+        foreach(i; 0 .. 3)
+        {
+            auto filename = genTempName(prefix, suffix, dir);
+
+            // O_EXCL does not support symbolic links.
+            if(dirName(filename).isSymlink)
+                filename = buildPath(absolutePath(dirName(filename), baseName(filename)));
+
+            import std.internal.cstring : tempCString;
+            version(Posix)
+            {
+                import core.sys.posix.sys.stat, core.sys.posix.fcntl;
+                enum flags = S_IRUSR | S_IWUSR;
```

Comment:
so then insert a return after the comma, there's no shame in

    auto fd = open(tempCString(filename), O_RDWR | O_CREAT | O_EXCL, 
        S_IRUSR | S_IWUSR);